src/UI/Index/Keybindings.hs

```diff
@@ -46,8 +50,25 @@ focusSearch s = continue $ s
 
 displayMail :: AppState -> T.EventM Name (T.Next AppState)
 displayMail s = do
-    s' <- liftIO $ updateStateWithParsedMail s
-    continue $ s'
+    s' <- liftIO $ updateStateWithParsedMail s >>= updateReadState removeTag
+    continue s'
+
+updateReadState :: (NotmuchMail -> Text -> NotmuchMail) -> AppState -> IO AppState
+updateReadState op s =
+    case L.listSelectedElement (view (asMailIndex . miListOfMails) s) of
+        Just (_,m) ->
+            let newTag = view (asConfig . confNotmuch . nmNewTag) s
+                dbpath = view (asConfig . confNotmuch . nmDatabase) s
+            in do setNotmuchMailTags dbpath (op m newTag)
+                  >>= either
+                      (\err -> pure $ s & asError ?~ err)
+                      (pure . updateMailInList s)
```

Comment:
perhaps the following instead of `(>>=)`
```haskell
either (\err -> set asError (Just err) s) (updateMailInList s)
  <$> setNotmuchMailTags dbpath (op m newTag)
```

Reply:
Nice. Cheers.